examples/openid_connect_demo/src/views/header.rs

```diff
@@ -0,0 +1,179 @@
+use crate::{
+    oidc::{
+        authorize_url, email, exchange_refresh_token, init_oidc_client, log_out_url,
+        AuthRequestState, AuthTokenState, ClientState,
+    },
+    router::Route,
+    storage::{use_persistent_set, AuthRequestEntry, AuthTokenEntry},
+    FERMI_AUTH_REQUEST, FERMI_AUTH_TOKEN, FERMI_CLIENT,
+};
+use dioxus::prelude::*;
+use dioxus_router::prelude::{Link, Outlet};
+use fermi::*;
+use openidconnect::{OAuth2TokenResponse, TokenResponse};
+
+#[component]
+pub fn LogOut(cx: Scope) -> Element {
+    let fermi_client_read = use_read(cx, &FERMI_CLIENT);
+    let fermi_auth_token_write = use_set(cx, &FERMI_AUTH_TOKEN);
+    let auth_token_read = use_read(cx, &FERMI_AUTH_TOKEN);
+    cx.render(match fermi_client_read.oidc_client.clone() {
+        Some(_client) => match auth_token_read.id_token.clone() {
+            Some(id_token) => {
+                let log_out_url_future =
+                    use_future(cx, (), |_| async move { log_out_url(id_token).await });
```

Comment:
[You cannot use hooks inside of a conditionals](https://dioxuslabs.com/learn/0.4/reference/hooks#no-hooks-in-conditionals) (the git version of `dx check` should catch this)